--- a/lib/util/charset/iconv.c
+++ b/lib/util/charset/iconv.c
@@ -31,7 +31,7 @@
 #include <unicode/utrans.h>
 #endif
 
-#if defined(ANDROID) && (!defined(__ANDROID_API__) || __ANDROID_API__ < 28)
+#if defined(ANDROID) && (!defined(__ANDROID_API__) || __ANDROID_API__ < 28) && !defined(__BIONIC_SWAB_INLINE)
 #include <byteswap.h>
 #endif
 
@@ -759,7 +759,7 @@
 	return 0;
 }
 
-#if defined(ANDROID) && (!defined(__ANDROID_API__) || __ANDROID_API__ < 28)
+#if defined(ANDROID) && (!defined(__ANDROID_API__) || __ANDROID_API__ < 28) && !defined(__BIONIC_SWAB_INLINE)
 void swab(const void *from, void*to, ssize_t n)
 {
   ssize_t i;
